[quant][be] Clean up QAT tests in test_quantize_pt2e.py

andrewor14 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #108061
* __->__ #107991

Summary: This commit does 4 main things:

1. When verifying QAT numerics, automatically check both the
per tensor and the per channel cases, and automatically verify
convert numerics

2. When verifying the QAT graph, automatically check both the
per tensor and the per channel cases

3. Merge verify graph and verify numerics tests for conv-bn

4. Fix `test_prepare_qat_conv_bn_fusion_getitem_placeholder`,
which was no longer testing the right thing recent capture
changes, since the maxpool op is no longer followed by a
getitem node. However, we do still need this test for other
ops that *are* followed by getitem nodes (e.g. standalone BN).

Items (1) - (3) make the QAT tests significantly less verbose
and easier to read.

Test Plan:
python test/test_quantization.py TestQuantizePT2E
python test/test_quantization.py TestQuantizePT2EModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

andrewor14 committed:
[quant][be] Clean up QAT tests in test_quantize_pt2e.py

Summary: This commit does 4 main things:

1. When verifying QAT numerics, automatically check both the
per tensor and the per channel cases, and automatically verify
convert numerics

2. When verifying the QAT graph, automatically check both the
per tensor and the per channel cases

3. Merge verify graph and verify numerics tests for conv-bn

4. Fix `test_prepare_qat_conv_bn_fusion_getitem_placeholder`,
which was no longer testing the right thing recent capture
changes, since the maxpool op is no longer followed by a
getitem node. However, we do still need this test for other
ops that *are* followed by getitem nodes (e.g. standalone BN).

Items (1) - (3) make the QAT tests significantly less verbose
and easier to read.

Test Plan:
python test/test_quantization.py TestQuantizePT2E
python test/test_quantization.py TestQuantizePT2EModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

jerryzh168 commented on test/quantization/pt2e/test_quantize_pt2e.py:
```diff
@@ -1928,57 +1936,25 @@ def forward(self, x):
+            m1, example_inputs, has_relu=False, has_bias=False,
         )
+        self._verify_symmetric_xnnpack_qat_numerics(m1, example_inputs)
+        self._verify_symmetric_xnnpack_qat_numerics(M2(), example_inputs)
```
should this be in a different test?

andrewor14 replied:
Maybe, but I'm not sure if I want to create a test just for this case so maybe I'll just leave this here for now

jerryzh168 commented on test/quantization/pt2e/test_quantize_pt2e.py:
```diff
@@ -2109,12 +2061,7 @@ def forward(self, x):
-        self._verify_symmetric_qnnpack_qat_numerics(
-            M(), example_inputs, is_per_channel=True, verify_convert=True,
-        )
+        self._verify_symmetric_xnnpack_qat_numerics(M(), example_inputs)
```
nit: M() is instantiated in 2062, maybe remove one of them

jerryzh168 reviewed (approved):
lg, thanks!

jerryzh168 reviewed (commented):
also might be helpful to move qat tests to a separate TestCase I think

andrewor14 committed:
Update on "[quant][be] Clean up QAT tests in test_quantize_pt2e.py"


Summary: This commit does 4 main things:

1. When verifying QAT numerics, automatically check both the
per tensor and the per channel cases, and automatically verify
convert numerics

2. When verifying the QAT graph, automatically check both the
per tensor and the per channel cases

3. Merge verify graph and verify numerics tests for conv-bn

4. Fix `test_prepare_qat_conv_bn_fusion_getitem_placeholder`,
which was no longer testing the right thing recent capture
changes, since the maxpool op is no longer followed by a
getitem node. However, we do still need this test for other
ops that *are* followed by getitem nodes (e.g. standalone BN).

Items (1) - (3) make the QAT tests significantly less verbose
and easier to read.

Test Plan:
python test/test_quantization.py TestQuantizePT2E
python test/test_quantization.py TestQuantizePT2EModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 committed:
Update on "[quant][be] Clean up QAT tests in test_quantize_pt2e.py"


Summary: This commit does 4 main things:

1. When verifying QAT numerics, automatically check both the
per tensor and the per channel cases, and automatically verify
convert numerics

2. When verifying the QAT graph, automatically check both the
per tensor and the per channel cases

3. Merge verify graph and verify numerics tests for conv-bn

4. Fix `test_prepare_qat_conv_bn_fusion_getitem_placeholder`,
which was no longer testing the right thing recent capture
changes, since the maxpool op is no longer followed by a
getitem node. However, we do still need this test for other
ops that *are* followed by getitem nodes (e.g. standalone BN).

Items (1) - (3) make the QAT tests significantly less verbose
and easier to read.

Test Plan:
python test/test_quantization.py TestQuantizePT2E
python test/test_quantization.py TestQuantizePT2EModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 commented:
> also might be helpful to move qat tests to a separate TestCase I think

Actually I think I'll do this in a separate PR so the code diff is still human legible

andrewor14 committed:
Update on "[quant][be] Clean up QAT tests in test_quantize_pt2e.py"


Summary: This commit does 4 main things:

1. When verifying QAT numerics, automatically check both the
per tensor and the per channel cases, and automatically verify
convert numerics

2. When verifying the QAT graph, automatically check both the
per tensor and the per channel cases

3. Merge verify graph and verify numerics tests for conv-bn

4. Fix `test_prepare_qat_conv_bn_fusion_getitem_placeholder`,
which was no longer testing the right thing recent capture
changes, since the maxpool op is no longer followed by a
getitem node. However, we do still need this test for other
ops that *are* followed by getitem nodes (e.g. standalone BN).

Items (1) - (3) make the QAT tests significantly less verbose
and easier to read.

Test Plan:
python test/test_quantization.py TestQuantizePT2E
python test/test_quantization.py TestQuantizePT2EModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 committed:
Update on "[quant][be] Clean up QAT tests in test_quantize_pt2e.py"


Summary: This commit does 4 main things:

1. When verifying QAT numerics, automatically check both the
per tensor and the per channel cases, and automatically verify
convert numerics

2. When verifying the QAT graph, automatically check both the
per tensor and the per channel cases

3. Merge verify graph and verify numerics tests for conv-bn

4. Fix `test_prepare_qat_conv_bn_fusion_getitem_placeholder`,
which was no longer testing the right thing recent capture
changes, since the maxpool op is no longer followed by a
getitem node. However, we do still need this test for other
ops that *are* followed by getitem nodes (e.g. standalone BN).

Items (1) - (3) make the QAT tests significantly less verbose
and easier to read.

Test Plan:
python test/test_quantization.py TestQuantizePT2E
python test/test_quantization.py TestQuantizePT2EModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

[ghstack-poisoned]

andrewor14 commented:
@pytorchbot merge
